proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -22,12 +22,64 @@ message CompleteValidateAfterOperation {
   double searcher_metric = 2;
 }
 
+// Constant Hyperparameter
+message ConstantHyperparameter {
+  // value of the constant hyperparameter.
+  double val = 1;
+}
+
+// Hyperparameter.
+message Hyperparameter {
+  // A user can provide one of the hyperparameter types for
+  // the custom searcher.
+  oneof union {
+    // Constant hyperparameter.
+    ConstantHyperparameter constant_hyperparam = 1;
+  }
+}
+
+// Create a trial with given hyperparameters.
+message CreateTrialOperation {
+  // The key refers to which part of the model this hyperparameters are for.
+  // An example in this map would be: {"optimizer": {"learning_rate": 0.01}}
+  map<string, Hyperparameter> hyperparams = 1;
```

Comment:
We support nested hyperparameters, which this doesn't cover. I suppose that can wait, though. But also the comment above seems to indicate a nested hyperparameter.

Reply:
Yup will add this later. Will remove the comment.